patches/api/0460-add-rich-message-component-support-to-configuration.patch

```diff
@@ -0,0 +1,85 @@
+From: Yannick Lamprecht <yannicklamprecht@live.de>
+Date: Tue, 6 Feb 2024 23:02:15 +0100
+Subject: [PATCH] add rich message component support to configuration
+
+
+diff --git a/src/main/java/org/bukkit/configuration/ConfigurationSection.java b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+index b6b00af08f12f838411845e4f4e29e62826dfc7f..5635c1fb01106e4f26a6d031db7872348be79e34 100644
+--- a/src/main/java/org/bukkit/configuration/ConfigurationSection.java
++++ b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+@@ -1058,4 +1058,54 @@ public interface ConfigurationSection {
+      * one line.
+      */
+     public void setInlineComments(@NotNull String path, @Nullable List<String> comments);
++
++    // Paper start - add rich message component support to configuration
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @return Requested Component.
++     */
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path) {
++        return getRichMessage(path, null);
++    }
++
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param component component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, !null -> !null")
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path, @Nullable net.kyori.adventure.text.Component component){
++        return getComponent(path, net.kyori.adventure.text.minimessage.MiniMessage.miniMessage(), component);
++    }
++
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param componentSerializer componentSerializer instance used for deserialization
++     * @param component component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, _, !null -> !null")
++    @Nullable
++    <C extends net.kyori.adventure.text.Component> C getComponent(@NotNull String path, @NotNull net.kyori.adventure.text.serializer.ComponentSerializer<C, C, String> componentSerializer, @Nullable C component);
```

Comment:
Do we care about a fallback-lacking overload? `getComponent(String, ComponentSerializer)`

Reply:
Addressed in https://github.com/PaperMC/Paper/commit/88393c02489850bbf8d582bf0da56bdea82fd408#diff-641780de4795f475f2b5c979f77ab04d7be650885ee99dd34b574ea6a0209983R74-R76